engine.go

```diff
@@ -167,30 +173,77 @@ func (e *Engine) Query(
 	return analyzed.Schema(), iter, nil
 }
 
+// ParseDefaults takes in a schema, along with each column's default value in a string form, and returns the schema
+// with the default values parsed and resolved.
+func ResolveDefaults(tableName string, schema []*ColumnWithRawDefault) (sql.Schema, error) {
+	ctx := sql.NewEmptyContext()
+	e := NewDefault()
+	db := memory.NewDatabase("temporary")
```

Comment:
This is no bueno

Reply:
There has to be a better way to do this than spin up a temporary in mem database

Reply:
`plan.DummyResolvedDB` it is lol. Since technically the memory database worked I went with it, but all of this stuff is just ugly to me. I'm not happy with the approach, but it can always be changed later (just that who knows when that'll be).